game-core/src/main/java/games/strategy/triplea/oddsCalculator/ta/AggregateResults.java

```diff
@@ -163,36 +151,27 @@ public double getAttackerWinPercent() {
     if (results.isEmpty()) {
       return 0.0;
     }
-    double count = 0;
-    for (final BattleResults result : results) {
-      if (result.attackerWon()) {
-        count++;
-      }
-    }
-    return count / results.size();
+    return results.stream()
+        .filter(BattleResults::attackerWon)
+        .count() / (double) results.size();
   }
 
   double getDefenderWinPercent() {
     if (results.isEmpty()) {
       return 0.0;
     }
-    double count = 0;
-    for (final BattleResults result : results) {
-      if (result.defenderWon()) {
-        count++;
-      }
-    }
-    return count / results.size();
+    return results.stream()
+        .filter(BattleResults::defenderWon)
+        .count() / (double) results.size();
   }
 
   public double getAverageBattleRoundsFought() {
     if (results.isEmpty()) {
       return 0.0;
     }
-    double count = 0;
-    for (final BattleResults result : results) {
-      count += result.getBattleRoundsFought();
-    }
+    final double count = results.stream()
+        .mapToInt(BattleResults::getBattleRoundsFought)
+        .sum();
```

Comment:
Since you're summing `int`s, should `count` be an `int`?  If so, then you'll have to cast `results.size()` below, as you've done elsewhere.